compat_test/compat_test.go

```diff
@@ -0,0 +1,1012 @@
+			if err != nil {
+				return fmt.Errorf("opEthclient.TransactionReceipt: %w", err)
+			}
+			results.opReceipts[index] = opReceipt
+			return nil
+		})
+
+		getTransactionReceipt := "eth_getTransactionReceipt"
+		g.Go(func() error {
+			err := rpcCall(ctx, clients.celoClient, &results.celoRawReceipts[index], getTransactionReceipt, hexHash)
+			if err != nil {
+				return fmt.Errorf("%s: %w", getTransactionReceipt, err)
+			}
+			return nil
+		})
+
+		g.Go(func() error {
+			err := rpcCall(ctx, clients.opClient, &results.opRawReceipts[index], getTransactionReceipt, hexHash)
+			if err != nil {
+				return fmt.Errorf("%s: %w", getTransactionReceipt, err)
+			}
+			return nil
+		})
+	}
+
+	err = g.Wait()
+	if err != nil {
+		return err
+	}
+
+	resultChan <- results
+	return err
+}
+
+func jsonConvert(in, out any) error {
+	marshaled, err := json.Marshal(in)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(marshaled, out)
+}
+
+func rpcCall(ctx context.Context, cl *rpc.Client, result interface{}, method string, args ...interface{}) error {
+	err := cl.CallContext(ctx, result, method, args...)
+	if err != nil {
+		return err
+	}
+	// All other responses should be non null, but for block receipts since it's
+	// not linked to a transaction we have to call the endpoint to find out if
+	// one exists.
+	if method != "eth_getBlockReceipt" {
+		if result == "nil" {
+			return fmt.Errorf("response for %v %v should not be null", method, args)
+		}
+	}
+	return nil
+}
+
+func filterOpReceipt(receipt map[string]interface{}) {
+	// Delete effective gas price fields that are nil, on the celo side we do not add them.
+	v, ok := receipt["effectiveGasPrice"]
+	if ok && v == nil {
+		delete(receipt, "effectiveGasPrice")
+	}
+}
+
+func filterOpTx(tx map[string]interface{}) {
+	// Some txs on celo contain chainID all of them do on op, so we just remove it from both sides.
+	delete(tx, "chainId")
+	// Celo never returned yParity
+	delete(tx, "yParity")
+	// Since we unequivocally delete gatewayFee on the celo side we need to delete it here as well.
+	delete(tx, "gatewayFee")
+}
+
+func filterCeloTx(tx map[string]interface{}) {
+	// Some txs on celo contain chainID all of them do on op, so we just remove it from both sides.
+	delete(tx, "chainId")
+	// On the op side we now don't return ethCompatible when it's true, so we
+	// remove it from the celo response in this case.
+	txType := tx["type"].(string)
+	if txType == "0x0" && tx["ethCompatible"].(bool) {
+		delete(tx, "ethCompatible")
+	}
+	//It seems gateway fee is always added to all rpc transaction responses on celo because tx.GatewayFee returns 0 if
+	//it's not set,even ethcompatible ones, this is confusing so we have removed this in the op code, so we need to make
+	//sure the celo side matches.
+	delete(tx, "gatewayFee")
+	// Transaction at block 9476837 returns a nullGasPrice on celo but the op-geth side returns the gasFeeCap
+	// delete(tx, "gasPrice")
```

Comment:
So does the script fail with this when run, or is this comment outdated?

Reply:
Outdated, thanks!